conda-store-server/conda_store_server/server/auth.py

```diff
@@ -233,31 +236,33 @@ def is_subset_entity_permissions(self, entity, new_entity):
                 return False
         return True
 
-    def authorize(self, entity, arn, required_permissions):
+    def authorize(
+        self, entity: schema.AuthenticationToken, arn: str, required_permissions
+    ):
         return required_permissions <= self.get_entity_permissions(
             entity=entity, arn=arn
         )
 
-    def database_role_bindings(self, entity):
+    def database_role_bindings(self, entity: schema.AuthenticationToken):
         with self.authentication_db() as db:
-            result = db.execute(
-                text(
-                    """
-                    SELECT nrm.entity, nrm.role
-                    FROM namespace n
-                    RIGHT JOIN namespace_role_mapping nrm ON nrm.namespace_id = n.id
-                    WHERE n.name = :primary_namespace
-                    """
-                ),
-                {"primary_namespace": entity.primary_namespace},
-            )
-            raw_role_mappings = result.mappings().all()
-
-            db_role_mappings = defaultdict(set)
-            for row in raw_role_mappings:
-                db_role_mappings[row["entity"]].add(row["role"])
-
-        return db_role_mappings
+            # Must have the same format as authenticated_role_bindings:
+            # {
+            #     "default/*": {"viewer"},
+            #     "filesystem/*": {"viewer"},
+            # }
+            res = defaultdict(set)
+
+            # FIXME: Remove try-except.
+            # Used in tests to check default permissions without populating the
+            # DB, which raises an exception since namespace is not found.
+            try:
+                roles = api.get_other_namespace_roles(db, name=entity.primary_namespace)
```

Comment:
We use `other` here because we're trying to find which namespaces we have access to. The non-other one is the opposite - which namespaces have access to us.